validator_client/src/block_service.rs

```diff
@@ -335,81 +335,235 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
             )
         }
 
-        for validator_pubkey in proposers {
-            let builder_proposals = self
-                .validator_store
-                .get_builder_proposals(&validator_pubkey);
-            let service = self.clone();
-            let log = log.clone();
-            self.inner.context.executor.spawn(
-                async move {
-                    if builder_proposals {
```

Comment:
Does this mean the `builder-proposal` flag in the VC is no longer relevant post-deneb? i.e. VC no longer has control over whether it uses builder proposals? If that's the case, we probably would have to deprecate the flag when we deprecate the v2 endpoint, and add a comment to the flag description that this flag is ignored when `produce-block-v3` flag is present or from Deneb.

Reply:
I added some comments to both `produce-block-v3 ` and `builder-proposal` flag descriptions as well as some TODOs that can be resolved post Deneb